Testing

smcommits opened the pull request:
This is the pull request for the testing branch of the Tic-Tac-Toe game
In this branch, we have created the test cases for every public method.

The public methods are contained within the following classes

  - Game
  - Player
  - Board

smcommits committed:
Fix merger conflicts

Nkuria committed:
Created tests for validate_input, validate_name and the player class

mimipeshy commented:
## STATUS : CHANGES REQUIRED :red_circle: 
Hey @smcommits  and @Nkuria :wave:  :wave: , you guys did an amazing job in testing this project, all tests pass. However let's make one more change, this will make you a better tester so don't get disappointed :smile: 

- [ ] A bit of common sense: check if specs cover and **negative scenarios** example we can test  `if a user does not enter a number between 1 and 9` should raise an error

That's all :relaxed:  , please submit another review when done.

Regards
Peris Ndanu
If you need further clarification, you can reach out to me on slack (peris ndanu)

smcommits committed:
Implement change request

Update the test condition for following methods to have a test for opposite scenario where it has
to be false
- valid_move?
- position_taken?
- valid_input?

smcommits committed:
Implement change request

smcommits commented:
@mimipeshy Thank you so much for the feedback!
I have implented the negative scenarios wherever applicable. For instance, in methods like _valid_move?_  and _position_taken?_. They will now return false if the condition is not true. 
Please feel free to provide any further feedback.

mimipeshy commented:
## STATUS :APPROVED :champagne:  :partying_face: 

Hey team :wave: , :wave:  congratulations on completing this milestone, the project is approved :smile: , go ahead and merge. All the best in the next milestones :100: 

Regards
Peris Ndanu
